apps/blockchain/lib/blockchain/transaction.ex

```diff
@@ -211,19 +211,49 @@ defmodule Blockchain.Transaction do
 
     {expended_gas, refund} = calculate_gas_usage(tx, remaining_gas, sub_state)
 
+    {account_interface_after_receipt, receipt} =
+      if empty_contract_creation?(tx) &&
+           Configuration.for(chain.evm_config).clean_touched_accounts?(chain.evm_config) do
+        account_interface_after_execution = AccountInterface.commit(updated_account_interface)
```

Comment:
It looks like there are some order dependence stuff here that I'm not quite sure I understand.

Here we're creating the account first, creating the receipt and then doing clean up and refunds.

In the other branch it looks like we clean up accounts and refunds then commit the account? After that we make the receipt.

Does the receipt depend on the account state? Should we always be creating the receipt at the end? Is it important that the account is created first in the empty contract case?